brvs-core/src/main/java/iroha/validation/transactions/plugin/impl/SoraDistributionPluggableLogic.java

```diff
@@ -0,0 +1,427 @@
+  private final String infoSetterAccount;
+
+  public SoraDistributionPluggableLogic(
+      QueryAPI queryAPI,
+      String projectAccounts,
+      String infoSetterAccount) {
+    Objects.requireNonNull(queryAPI, "Query API must not be null");
+    if (StringUtils.isEmpty(projectAccounts)) {
+      throw new IllegalArgumentException("Project accounts must not be neither null nor empty");
+    }
+    if (StringUtils.isEmpty(infoSetterAccount)) {
+      throw new IllegalArgumentException("Info setter account must not be neither null nor empty");
+    }
+
+    this.queryAPI = queryAPI;
+    this.brvsAccountId = queryAPI.getAccountId();
+    this.brvsKeypair = queryAPI.getKeyPair();
+    this.projectAccounts = new HashSet<>(Arrays.asList(projectAccounts.split(COMMA_SPACES_REGEX)));
+    this.infoSetterAccount = infoSetterAccount;
+
+    logger.info("Started distribution processor with project accounts: {}", this.projectAccounts);
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  public Map<String, BigDecimal> filterAndTransform(Iterable<Transaction> sourceObjects) {
+    // sums all the xor transfers per project owner
+    return StreamSupport.stream(sourceObjects.spliterator(), false)
+        .flatMap(
+            transaction -> transaction.getPayload()
+                .getReducedPayload()
+                .getCommandsList()
+                .stream()
+        )
+        .filter(Command::hasTransferAsset)
+        .map(Command::getTransferAsset)
+        .filter(command -> XOR_ASSET_ID.equals(command.getAssetId()) &&
+            projectAccounts.contains(command.getSrcAccountId()))
+        .collect(
+            Collectors.groupingBy(
+                TransferAsset::getSrcAccountId,
+                Collectors.reducing(
+                    BigDecimal.ZERO,
+                    transfer -> new BigDecimal(transfer.getAmount()),
+                    BigDecimal::add
+                )
+            )
+        );
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  @Override
+  protected void applyInternal(Map<String, BigDecimal> processableObject) {
+    processDistributions(processableObject);
+  }
+
+  /**
+   * Processes committed project owners transfers and performs corresponding distributions if
+   * needed
+   *
+   * @param transferAssetMap {@link Map} of project owner account id to aggregated volume of their
+   * transfer within the block
+   */
+  private void processDistributions(Map<String, BigDecimal> transferAssetMap) {
+    // list for batches to send after processing
+    final List<Transaction> transactionList = new ArrayList<>();
+    transferAssetMap.forEach((projectOwnerAccountId, transferAmount) -> {
+      logger.info("Triggered distributions for {}", projectOwnerAccountId);
+      final SoraDistributionFinished distributionFinished = queryDistributionsFinishedForAccount(
+          projectOwnerAccountId
+      );
+      if (distributionFinished != null
+          && distributionFinished.finished != null
```

Comment:
Maybe make `SoraDistributionFinished.finished ` primitive?

Reply:
Not sure how absence of the field would be handled by gson while deserializing since it couldn't become `null`